Spigot-Server-Patches/0356-Add-source-block-to-BlockPhysicsEvent.patch

```diff
@@ -0,0 +1,22 @@
+From 28343c946f259516a859770bc25556322d243d96 Mon Sep 17 00:00:00 2001
+From: Sotr <i@omc.hk>
+Date: Wed, 22 Aug 2018 17:20:46 +0800
+Subject: [PATCH] Add source block to BlockPhysicsEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
+index 04d0fa1..523dd22 100644
+--- a/src/main/java/net/minecraft/server/World.java
++++ b/src/main/java/net/minecraft/server/World.java
+@@ -590,7 +590,7 @@ public abstract class World implements IBlockAccess {
+                 // CraftBukkit start
+                 CraftWorld world = ((WorldServer) this).getWorld();
+                 if (world != null && !((WorldServer)this).stopPhysicsEvent) { // Paper
+-                    BlockPhysicsEvent event = new BlockPhysicsEvent(world.getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ()), CraftMagicNumbers.getId(block));
++                    BlockPhysicsEvent event = new BlockPhysicsEvent(world.getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ()), CraftMagicNumbers.getId(block), world.getBlockAt(blockposition1.getX(), blockposition1.getY(), blockposition1.getZ())); // Paper - add source block
```

Comment:
This will destroy server performance even more.

Modify this to only pass the x/y/z to the BPE, and only call getBlockAt if someone calls BPE.getSourceBlock

Reply:
Good idea, what about also making `getBlock` lazy-loading? should I open a new pr?